tests/integration-tests.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+# This runs all integration test in isolation
+
+set -ex
+
+# Ensure current path is project root
+cd "$(dirname "$0")/../"
+
+QDRANT_HOST='localhost:6333'
+
+# Build
+$(cargo build --features grpc)
+# Run in background
+$(./target/debug/qdrant) &
+
+## Capture PID of the run
+PID=$(pidof "./target/debug/qdrant")
+echo $PID
+
+until $(curl --output /dev/null --silent --get --fail http://$QDRANT_HOST/collections); do
+  printf 'waiting for server to start...'
+  sleep 5
+done
+
+echo "server ready to serve traffic"
+
+./tests/basic_api_test.sh
+
+./tests/basic_grpc_test.sh
+
+echo "server is going down"
+$(kill -9 $PID)
```

Comment:
the server does not seem to go down when asked gently with `kill -15`

Reply:
As far as I know GitHub Actions will kill the background process at the end of the job by themselves. This is due to the fact that jobs are ran in isolated containers. So basically when GitHub Actions reaches the end of the steps and the last command completes it will just drop the container.

Reply:
Though I could not find it in their docs explicitly.

Reply:
Ok, found this:
> If you use a GitHub-hosted runner, each job runs in a fresh instance of a virtual environment specified by runs-on.
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#choosing-github-hosted-runners

And we use GitHub hosted runners, so it should be fine.

Reply:
I decided the shutdown the server in the script to avoid having to kill the server manually when using the script locally.

I expect we will have to run this script locally from to time to time for debugging purpose therefore I think it would makes sense to keep the shutdown feature.

WDYT?

Reply:
Ok, sure. I thought it was only for CI.